pkgs/by-name/wh/whatsie/package.nix

```diff
@@ -0,0 +1,58 @@
+{
+  fetchFromGitHub,
+  lib,
+  stdenv,
+  libX11,
+  libxcb,
+  qt5,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "whatsie";
+  version = "4.15.3";
+
+  src = fetchFromGitHub {
+    owner = "keshavbhatt";
+    repo = "whatsie";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-fbbyh8Q1jbKY+4ZEexDvIrhdcP+uCYAnhtUQsSqS0jw=";
+  };
+
+  sourceRoot = "${finalAttrs.src.name}/src";
+
+  buildInputs = [
+    libX11
+    libxcb
+    qt5.qtwebengine
+  ];
+
+  nativeBuildInputs = [
+    qt5.wrapQtAppsHook
+    qt5.qmake
+  ];
+
+  strictDeps = false;
+
+  enableParallelBuilding = true;
+
+  preBuild = ''
+    export QT_WEBENGINE_ICU_DATA_DIR=${qt5.qtwebengine.out}/resources
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 whatsie -t $out/bin
+
+    runHook postInstall
+  '';
+
```

Comment:
I was playing with something similar, when I was experimenting with `installCheckPhase`. Unfortunatelly, qt couldn't connect to display properly and fails. Similar thing happens here:

```
       > whatsie --version returned a non-zero exit code.
       > qt.qpa.xcb: could not connect to display
       > qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
       > This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
       >
       > Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
```

Reply:
Hum... maybe it requires X11 (or Wayland or anything similar). Let's forget this for now.